### ISSUE TITLE
update `no-unused-components` to not mix pascal/camel casing

### ISSUE THREAD

**Issue by evykassirer** [vuejs/eslint-plugin-vue#1445]
**What rule do you want to change?**

no-unused-components

**Does this change cause the rule to produce more or fewer warnings?**

more (but for things that are actively broken)

**How will the change be implemented? (New option, new default behavior, etc.)?**

Update the lines around here https://github.com/vuejs/eslint-plugin-vue/blob/master/lib/rules/no-unused-components.js#L108 to not mix pascal and camel case

**Please provide some example code that this change will affect:**

```vue
<template>
    <MyCoolComponent />
</template>

<script>
import Vue from "vue";

import myCoolComponent from "path/to/my-cool-component.vue";

export default Vue.extend({
    name: "ParentComponent",

    components: {
        myCoolComponent,
    },

    ...
```

```vue
<template>
    <myCoolComponent />
</template>

<script>
import Vue from "vue";

import MyCoolComponent from "path/to/my-cool-component.vue";

export default Vue.extend({
    name: "ParentComponent",

    components: {
        MyCoolComponent,
    },

    ...
```

**What does the rule currently do for this code?**

it doesn't fail on this code, but it's causing us Vue runtime errors (the component doesn't load and there's an unregistered component error)

**What will the rule do after it's changed?**

it will label the component as not registered

**Additional context**

note that these work fine

* `my-cool-component` in the template and `myCoolComponent` in the registered components list
* `my-cool-component` in the template and `MyCoolComponent` in the registered components list 

but these cause Vue runtime errors

* `MyCoolComponent` in the template and `myCoolComponent` in the registered components list 
* `myCoolComponent` in the template and `MyCoolComponent` in the registered components list 

curious if I'm missing some context on how Vue works? or does the linting rule need to be updated?

**Comment by ota-meshi** [comment]
Thank you for this issue!
It would be great if we could catch the cause of the runtime error in advance.
